w02/test/gad/binarysearch/BoundSearchTest.java

```diff
@@ -0,0 +1,93 @@
+				Arguments.of(new int[]{1, 2, 2, 4, 4, 4, 4, 8, 8, 27, 89}, 4, true, 3),
+				Arguments.of(new int[]{1, 2, 2, 4, 4, 4, 4, 8, 8, 27, 89}, 4, false, 6),
+				Arguments.of(new int[]{1, 2, 2, 4, 4, 4, 4, 8, 8, 27, 89}, 8, true, 7),
+				Arguments.of(new int[]{1, 2, 2, 4, 4, 4, 4, 8, 8, 27, 89}, 8, false, 8)
+		);
+	}
+
+	static Stream<Arguments> valueNotInArray() {
+		return Stream.of(
+				Arguments.of(new int[]{1, 2, 3, 4}, 5, true, -1),
+				Arguments.of(new int[]{1, 2, 3, 4}, 5, false, 3),
+				Arguments.of(new int[]{1, 2, 3, 4}, 0, true, 0),
+				Arguments.of(new int[]{1, 2, 2, 2, 6, 7, 8, 9, 10}, 0, false, -1),
+				Arguments.of(new int[]{1, 2, 2, 2, 6, 7, 8, 9, 10}, -1, true, 0),
+				Arguments.of(new int[]{1, 2, 2, 4, 4, 4, 4, 8, 8, 27, 89}, 3, true, 3),
+				Arguments.of(new int[]{1, 2, 2, 4, 4, 4, 4, 8, 8, 27, 89}, 3, false, 2),
+				Arguments.of(new int[]{1, 2, 2, 4, 4, 4, 4, 8, 8, 27, 89}, 7, true, 7),
+				Arguments.of(new int[]{1, 2, 2, 4, 4, 4, 4, 8, 8, 27, 89}, 7, false, 6)
+		);
+	}
+
+	static Stream<Arguments> randomValues() {
+		return Stream.of(
+				Arguments.of(new int[]{0, 1, 1, 1, 1, 1, 1, 1}, 1, true, 1),
+				Arguments.of(new int[]{1, 1, 1, 1, 1, 1, 1}, 2, true, -1),
+				Arguments.of(new int[]{1, 1, 1, 1, 1, 1, 1, 4}, 1, false, 6),
+				Arguments.of(new int[]{1, 4, 4, 4, 5, 5, 5, 5, 8}, 4, true, 1),
+				Arguments.of(new int[]{1, 4, 4, 4, 5, 5, 5, 5, 8}, 5, true, 4),
+				Arguments.of(new int[]{1, 4, 4, 4, 5, 5, 5, 5, 8}, 5, false, 7),
+				Arguments.of(new int[]{1, 4, 4, 4, 5, 5, 5, 5, 8}, 9, false, 8),
+				Arguments.of(new int[]{1, 4, 4, 4, 5, 5, 5, 5, 8, 10, 23}, 6, true, 8),
+				Arguments.of(new int[]{1, 4, 4, 4, 5, 5, 5, 5, 8, 10, 23}, 6, false, 7),
+				Arguments.of(new int[]{1, 4, 4, 4, 5, 5, 5, 5, 8}, 9, true, -1),
+				Arguments.of(new int[]{1, 2, 2, 4, 4, 4, 4, 8, 8, 27, 89}, 25, true, 9),
+				Arguments.of(new int[]{1, 2, 2, 4, 4, 4, 4, 8, 8, 27, 89}, 25, false, 8),
+				Arguments.of(new int[]{1, 2, 2, 4, 4, 4, 4, 8, 8, 27, 89}, 99, true, -1),
+				Arguments.of(new int[]{1, 2, 2, 4, 4, 4, 4, 8, 8, 27, 89}, 99, false, 10),
+				Arguments.of(new int[]{1, 2, 2, 4, 4, 4, 4, 8, 8, 27, 89}, -1, true, 0),
+				Arguments.of(new int[]{1, 2, 2, 4, 4, 4, 4, 8, 8, 27, 89}, -1, false, -1)
+		);
+	}
+
+	static Stream<Arguments> sameValues() {
+		return Stream.of(
+				Arguments.of(new int[]{1, 1, 1, 1, 1, 1, 1}, 1, true, 0),
+				Arguments.of(new int[]{1, 1, 1, 1, 1, 1, 1}, 1, false, 6)
+		);
+	}
+
+	static Stream<Arguments> valuesOnArrayBoundaries() {
+		return Stream.of(
+				Arguments.of(new int[]{0, 4, 4, 4, 4, 9, 10}, 2, true, 1),
+				Arguments.of(new int[]{0, 4, 4, 4, 4, 9, 10}, 11, true, -1),
+				Arguments.of(new int[]{0, 4, 4, 4, 4, 9, 10}, 11, false, 6),
+				Arguments.of(new int[]{1, 1, 4, 4, 4, 9, 10}, 1, false, 1)
+		);
+	}
+
+	@ParameterizedTest
+	@MethodSource({"simple", "randomValues", "sameValues", "valueNotInArray", "duplicatesInArray", "valuesOnArrayBoundaries"})
```

Comment:
I have run the tests locall and I pass all, but for better understanding which Argument is running could you please bringt it in the order of the MethodSource if it' possible for you.

Reply:
Fixed in [a4d01c2](https://github.com/JohannesStoehr/gad23-tests/pull/18/commits/a4d01c22db9f840e1fd8c35a42fbcb3b9710f5ab)